### PR TITLE
Add meta description and squish content

### PR DESCRIPTION
**Pull request by salim-b**
- Adds a `<meta name="description">` tag to the generated HTML (with the same content as the `<meta property="og:description">` tag) which [pleases Lighthouse](https://web.dev/meta-description/).
- Normalizes whitespace chars within content extracted from the DESCRIPTION file fields (like `Title` and `Description`).

### CODE REVIEW

**Comment by hadley** [comment]
Do you want to finish this one off next? It'll need a rebase/merge to begin.

**Comment by hadley** [comment]
And worth checking if https://github.com/r-lib/desc/blob/master/NEWS.md#breaking-change resolves the squishing issue.

**Comment by salim-b** [comment]
> And worth checking if https://github.com/r-lib/desc/blob/master/NEWS.md#breaking-change resolves the squishing issue.

Yeah, it does (I've submitted that PR :smile: ). So it would be ok to require `desc >= 1.4.0` in DESCRIPTION?

**Comment by hadley** [comment]
@salim-b yup, that'd be fie.

**Comment by salim-b** [comment]
I've replaced all occurences of

```r
desc$get("<NAME>")[[1]]
```

with

```r
desc$get_field(
  key = "<NAME>",
  default = NA_character
)
```

and increased the min. desc pkg version to v1.4.0.
The `default = NA_character` arg ensures no unintended errors arise in case a `"<NAME>"` field is unset/missing.

File: R/build-home-index.R [hunk 1]
```diff
@@ -12,7 +12,10 @@ build_home_index <- function(pkg = ".", quiet = TRUE) {
   data <- data_home(pkg)
 
   if (is.null(src_path)) {
-    data$index <- linkify(pkg$desc$get("Description")[[1]])
+    data$index <- linkify(pkg$desc$get_field(
+      key = "Description",
+      default = NA_character_
```
Comment (hadley) [inline R/build-home-index.R:17]: I think a default value of `""` would be better, you don't need to spell out the argument names.

File: R/build-home-index.R [hunk 2]
```diff
@@ -38,10 +41,13 @@ data_home <- function(pkg = ".") {
 
   print_yaml(list(
     pagetitle = pkg$meta$home[["title"]] %||%
-      cran_unquote(pkg$desc$get("Title")[[1]]),
+      cran_unquote(pkg$desc$get_field(key = "Title", default = NA_character_)),
     sidebar = data_home_sidebar(pkg),
     opengraph = list(description = pkg$meta$home[["description"]] %||%
-                       cran_unquote(pkg$desc$get("Description")[[1]])),
+                       cran_unquote(pkg$desc$get_field(
```
Comment (hadley) [inline R/build-home-index.R:47]: Needs correct indenting.

File: R/build-reference.R [hunk 3]
```diff
@@ -348,7 +348,7 @@ data_reference_topic <- function(topic,
 
   # Sections that contain arbitrary text and need cross-referencing
   out$description <- as_data(tags$tag_description[[1]])
-  out$opengraph <- list(description = strip_html_tags(out$description$contents))
+  out$opengraph <- list(description = str_squish(strip_html_tags(out$description$contents)))
```
Comment (hadley) [inline R/build-reference.R:351]: Isn't the squishing now done by desc?

**Review by hadley — COMMENTED** [review]
It's easier to review code if you keep changes more closely aligned to the purpose of the PR. Here the goal is to improve page descriptions, so I think it would be better to avoid changes not related to that goal.

**Comment by salim-b** [comment]
> It's easier to review code if you keep changes more closely aligned to the purpose of the PR. Here the goal is to improve page descriptions, so I think it would be better to avoid changes not related to that goal.

Ok, I understand. I just wanted to unify pkgdown's internal behaviour by always using `desc$get_field()` instead of `desc$get()` to retrieve single DESCRIPTION field values and thought it would really make sense to do it here while we're at it.

File: R/build-reference.R (hunk 3 above)
Comment (salim-b) [reply R/build-reference.R:355]: Nope, not in this case since the input here stems from parsing `.Rd` files.

**Pull request closed without merging**